reconstruction/ecoli/dataclasses/process/metabolism_new_pathway.py

```diff
@@ -0,0 +1,328 @@
+        '''
+        y, rates = self._make_y_dy()
+
+        rates = sp.Matrix(rates)
+        J = rates.jacobian(y)
+
+        self.symbolic_rates = rates
+        self.symbolic_rates_jacobian = J
+
+    def molecules_to_next_time_step(self, moleculeCounts, moleculeNames, enzymeNames, enzymeDict, cellVolume,
+                                    nAvogadro, timeStepSec, random_state, method="LSODA", min_time_step=None, jit=True, methods_tried=None):
+
+        """
+        Calculates the changes in the counts of molecules in the next timestep
+        by solving an initial value ODE problem.
+
+        Args:
+            moleculeCounts (1d ndarray, ints): current counts of molecules
+                involved in the ODE
+            cellVolume (float): current volume of cell
+            nAvogadro (float): Avogadro's number
+            timeStepSec (float): current length of timestep in seconds
+            method (str): name of the ODE method to use
+            min_time_step (int): if not None, timeStepSec will be scaled down until
+                it is below min_time_step if negative counts are encountered
+            jit (bool): if True, use the jit compiled version of derivatives
+                functions
+            methods_tried (Optional[Set[str]]): methods for the solver that have
+                already been tried
+
+        Returns:
+            moleculesNeeded (1d ndarray, ints): counts of molecules that need
+                to be consumed
+            allMoleculesChanges (1d ndarray, ints): expected changes in
+                molecule counts after timestep
+        """
+
+        enzymeC = []
+        for e in enzymeNames:
+            if e is None:
+                enzymeC.append(float(1))
+            else:
+                enzymeC.append(float(enzymeDict[e+'[c]']))
+
+        print('Molecule counts aloc', moleculeCounts)
+        y_init = moleculeCounts / (cellVolume * nAvogadro)
+        # TO DO Ioana: Fix the jit version. At the moment it is identical to the non-jit
+        if jit:
+            derivatives = self.derivatives_jit
+            derivatives_jacobian = self.derivatives_jacobian_jit
+        else:
+            derivatives = self.derivatives
+            derivatives_jacobian = self.derivatives_jacobian
+
+        sol = scipy.integrate.solve_ivp(
+            lambda t, y: derivatives([0, timeStepSec], y_init, enzymeC, moleculeNames), [0, timeStepSec], y_init,
+            method=method, t_eval=[0, timeStepSec], atol=1e-8)
+        y = sol.y.T
+
+
+        y[y < 0] = 0
+        yMolecules = y * (cellVolume * nAvogadro)
+
+
+        # Calculate changes in molecule counts for all molecules
+        allMoleculesChanges = yMolecules[-1, :] - yMolecules[0, :]
+
+        # Molecules needed are the change but only for the molecules corresponding to the negative terms in the stoich matrix
+        moleculesNeeded = abs(np.sum(self.stoich_matrix().clip(max=0), axis=1) * allMoleculesChanges)
```

Comment:
I'm not sure if I understand the calculation here.. is there a reason we need to deviate from the 2CS method of calculating the molecules needed by simply doing this?
```np.negative(allMoleculesChanges).clip(min=0)```